tests/class/class.cmake

```diff
@@ -144,42 +140,289 @@ ct_add_test("cpp_member_function_calls")
         endfunction()
     cpp_end_class()
 
-    ct_add_section("member functions with a single parameter can be called")
+    ct_add_section("Member functions with a single parameter can be called")
         MyClass(CTOR my_instance)
         MyClass(fxn_a "${my_instance}" 1)
         ct_assert_prints("a = 1")
     ct_end_section()
 
-    ct_add_section("member functions with multiple parameters can be called")
+    ct_add_section("Member functions with multiple parameters can be called")
         MyClass(CTOR my_instance)
         MyClass(fxn_b "${my_instance}" 1 2)
         ct_assert_prints("a = 1, b = 2")
     ct_end_section()
 
-    ct_add_section("member functions with desc parameters can be called")
+    ct_add_section("Member functions with desc parameters can be called")
         MyClass(CTOR my_instance)
         MyClass(fxn_c "${my_instance}" 1 "This is a string.")
         ct_assert_prints("a = 1, b = This is a string.")
     ct_end_section()
-
 ct_end_test()
 
 ct_add_test("cpp_attr")
     include(cmakepp_core/class/class)
 
-    cpp_class(MyClass)
-
     ct_add_section("Signature")
         set(CMAKEPP_CORE_DEBUG_MODE ON)
 
+        cpp_class(MyClass)
+
         ct_add_section("0th argument must be class")
             cpp_attr(TRUE an_attr)
             ct_assert_fails_as("Assertion: bool is convertible to class")
         ct_end_section()
+    ct_end_section()
+
+    ct_add_section("Can retrieve attributes")
+        # Declare class with functions for getting and printing attributes
+        cpp_class(MyClass)
+            cpp_attr(MyClass a 1)
+            cpp_attr(MyClass b 2)
+            cpp_attr(MyClass c 3)
+            cpp_attr(MyClass d 4)
+
+            # Get multiple attributes one at a time
+            cpp_member(get_single_attrs MyClass)
+            function("${get_single_attrs}" self)
+                MyClass(GET "${self}" _gsa_a a)
+                MyClass(GET "${self}" _gsa_d d)
+                message("${_gsa_a}, ${_gsa_d}")
+            endfunction()
+
+            # Get multiple attributes and return using prefix
+            cpp_member(get_multiple_attrs MyClass)
+            function("${get_multiple_attrs}" self)
+                MyClass(GET "${self}" _gma a b c d)
+                message("${_gma_a}, ${_gma_b}, ${_gma_c}, ${_gma_d}")
+            endfunction()
+        cpp_end_class()
+
+        # Create instance
+        MyClass(CTOR my_instance)
+
+        ct_add_section("Can retrieve a single attribute in one call")
+            MyClass(get_single_attrs "${my_instance}")
```

Comment:
It's probably better to just explicitly test the function, i.e.,
```cmake
MyClass(GET "${my_instance}" _gsa a)
ct_assert_equal(_gsa_a 1)
```
rather than writing a wrapper function and testing that the wrapper function prints. Printing in CMake is somewhat fickle (the `message` command has all sorts of internal formatting that can bite you).

This comment applies to the majority of the unit tests in this file.